app/models/statistics_announcement.rb

```diff
@@ -157,25 +162,10 @@ def unpublished?
     publishing_state == "unpublished"
   end
 
-  def publish_if_needed!
-    publish if !unpublished?
+  def requires_redirect?
+    unpublished? || !!publication.try(:published?)
```

Comment:
Sorry to nitpick but I'd prefer the old school `|| publication && publication.published?` in place of the `try` gymnastics. But don't let that be a blocker if you disagree.